src/main/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributor.java

```diff
@@ -39,129 +40,112 @@
 import uk.gov.ons.ctp.response.collection.exercise.representation.SampleUnitGroupDTO.SampleUnitGroupEvent;
 import uk.gov.ons.ctp.response.collection.exercise.representation.SampleUnitGroupDTO.SampleUnitGroupState;
 import uk.gov.ons.ctp.response.collection.exercise.service.EventService;
+import uk.gov.ons.ctp.response.party.representation.Association;
+import uk.gov.ons.ctp.response.party.representation.Enrolment;
+import uk.gov.ons.ctp.response.party.representation.PartyDTO;
+import uk.gov.ons.ctp.response.sample.representation.SampleUnitDTO;
 
 /** Class responsible for business logic to distribute SampleUnits. */
 @Component
 public class SampleUnitDistributor {
   private static final Logger log = LoggerFactory.getLogger(SampleUnitDistributor.class);
 
   private static final String DISTRIBUTION_LIST_ID = "group";
-  // this is a bit of a kludge - jpa does not like having an IN clause with an
-  // empty list
-  // it does not return results when you expect it to - so ... always have this
-  // in the list of excluded case ids
+  private static final String ENABLED = "ENABLED";
   private static final int IMPOSSIBLE_ID = Integer.MAX_VALUE;
   private static final int TRANSACTION_TIMEOUT = 60;
-  // single TransactionTemplate shared amongst all methods in this instance
-  private final TransactionTemplate transactionTemplate;
-  @Autowired private AppConfig appConfig;
-  @Autowired private EventRepository eventRepository;
-  @Autowired private SampleUnitGroupRepository sampleUnitGroupRepo;
-  @Autowired private SampleUnitRepository sampleUnitRepo;
-  @Autowired private CollectionExerciseRepository collectionExerciseRepo;
-  @Autowired private SampleUnitPublisher publisher;
-  @Autowired private SampleUnitDistributorHelper sampleUnitDistributorHelper;
-
-  @Autowired
-  @Qualifier("collectionExercise")
+
+  private AppConfig appConfig;
+
+  private SampleUnitPublisher publisher;
+
+  private CollectionExerciseRepository collectionExerciseRepo;
+  private EventRepository eventRepository;
+  private SampleUnitGroupRepository sampleUnitGroupRepo;
+  private SampleUnitRepository sampleUnitRepo;
+
+  private ActionSvcClient actionSvcClient;
+  private PartySvcClient partySvcClient;
+
   private StateTransitionManager<CollectionExerciseState, CollectionExerciseEvent>
       collectionExerciseTransitionState;
-
-  @Autowired
-  @Qualifier("sampleUnitGroup")
   private StateTransitionManager<SampleUnitGroupState, SampleUnitGroupEvent> sampleUnitGroupState;
 
-  @Autowired
-  @Qualifier("distribution")
   private DistributedListManager<Integer> sampleDistributionListManager;
+  private final TransactionTemplate transactionTemplate;
 
-  /**
-   * Constructor into which the Spring PlatformTransactionManager is injected
-   *
-   * @param transactionManager provided by Spring
-   */
-  @Autowired
-  public SampleUnitDistributor(final PlatformTransactionManager transactionManager) {
+  public SampleUnitDistributor(
```

Comment:
whoa! That is a LOT of dependencies. Clearly this is a GOD class which needs to be refactored